content/blog/stackstorm-simple-elegant-automation-for-everyone.md

```diff
@@ -0,0 +1,33 @@
+
+I like writing code, but you need to work smarter, not harder. After all, there is only so much time in the day. That is why I love using python. Python has a lot of pre-written modules that make it extremely extensible. By using the **import** command in my script I can take advantage of modules I have no idea how to even start writing. I like the **plug and play** of how easy it is to consume those extra modules. I like using **requests**, I can just code **import requests**, and I have access to a well developed chunk of code that I did not have to write.
+
+What if we applied the same concept to our problem at the start of this blog? What if each of the resources in the data center had a **module** of sorts, that I could plug into some sort of framework? Something like StackStorm!
+
+![](https://www.techworldwookie.com/images/stackstorm.png)
+
+Before we get too far, here is a quick re-cap of what StackStorm is. StackStorm uses Sensors, Rules, Triggers, Actions, and Workflows. You can think of it as If-this-then-that automation. Based on a event on some device, a StackStorm sensor can detect it, and this could trigger a Rule. The Rule would have some logic in it and if the trigger launched the Rule, the Rule would run an action, or a series of actions called a Workflow. It might sound a little involved but it is all packaged together to make it easy to deploy these solutions.
+
+![](https://www.techworldwookie.com/images/process.png)
+
+What if I told you there was a place called the StackStorm Exchange that already had 170 of these modules just waiting for you to plug into your projects? [Look at this link.](applewebdata://EBBB076B-5028-426E-9291-45124FE61477/exchange.stackstorm.com) It is quite an impressive stockpile of automation just waiting for you. I recently finished developing a StackStorm integration pack for HPE Primera Storage. Before that, I developed a pack to integrate HPE Nimble Storage as well. What communication can we develop? Let's take a look.
+
+I just needed an example of these two systems communicating. I settled on volumes. I wanted to be able to create a volume on the HPE Primera and have it **automatically** appear on a HPE Nimble storage array. Maybe this is something cool, maybe not, but it was just going to demonstrate the power of HPE devices talking to one another and have the capability to talk to the StackStorm integration packs out on the exchange. The process would look something like this:
+
+![](https://www.techworldwookie.com/images/flow.png)
+
+The most important thing to remember is that when I run a StackStorm workflow, one step in the process can be running a StackStorm action from one integration pack, and the next task can be a StackStorm action from any other pack! Now, I have StackStorm integration packs for HPE OneView, HPE Nimble, and HPE Primera. Not only can these three data center systems communicate with one another, they now have access to the StackStorm Exchange! 
```

Comment:
The last sentence is confusing me. Maybe modify the last sentence to: (not sure I'm understanding it correctly)
Using these integration packs, not only can these three data center systems communicate with one another, they can also access the StackStorm Exchange!

Reply:
Changed:  Using these StackStorm integration packs allows HPE OneView, HPE Nimble, and HPE Primera communicate with one another. Because they are part of StackStorm, they now have access to all the pre-written automation packs available on the StackStorm Exchange!